nominal/core/dataset.py

```diff
@@ -261,6 +261,8 @@ def set_channel_units(self, channels_to_units: Mapping[str, str | None], validat
             )
             update_requests.append(channel_request)
 
+        if not update_requests:
+            return
```

Comment:
if this fixes a bug, the PR should be a fix: instead. Alternatively you can split out this tiny fix into a tiny PR to merge first